aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx

```diff
@@ -190,16 +368,139 @@ class Pipeline extends Component {
 
   toggleFullscreen () {
     if (!this.state.fullscreen) {
-      this.setState({ fullscreen: true })
+      this.setState({
+        fullscreen: true,
+        showOutput: false
+      })
     } else {
       this.setState({ fullscreen: false })
     }
   }
+
+  viewInput () {
+    this.setState({
+      showOutput: false,
+      pipelineView: 'input',
+      showSettings: false
+    })
+  }
+
+  onSettingsSave () {
+    const { selectedContract } = this.props
+    if (!selectedContract.id) {
+      const newContract = { ...selectedContract, pipeline: this.props.selectedPipeline.id, name: this.state.contractName }
+      this.props.addInitialContract(newContract)
+      this.setState({
+        newContracts: []
+      })
+      this.renderNewContractTabs([])
+    } else {
+      this.props.updateContract({ ...selectedContract, name: this.state.contractName })
+    }
+    this.toggleSettings()
+  }
+
+  renderSettings () {
+    return (
+      <div className='pipeline-settings'>
+        <div
+          className='btn-icon close-button'
+          onClick={() => this.toggleSettings()}>
+          <i className='fas fa-times' />
+          <span>close settings</span>
+        </div>
+
+        <div className='contract-form'>
+          <div className='form-group'>
+            <label className='form-label'>
+              <FormattedMessage
+                id='contract.new.name'
+                defaultMessage='Contract name'
+              />
+            </label>
+            <input
+              type='text'
+              required
+              name='name'
+              className='input-d contract-name'
+              value={this.state.contractName}
+              onChange={e => this.setState({ contractName: e.target.value })}
+            />
+          </div>
+
+          <div className='settings-section'>
+            <label className='form-label'>
+              <FormattedMessage
+                id='settings.puplish.title'
+                defaultMessage='Publish status'
+              />
+            </label>
+            <div className='status-publish'>
+              <FormattedMessage
+                id='pipeline.publish-status'
+                defaultMessage={this.props.selectedContract.published_on
+                  ? `Published on ${moment(this.props.selectedContract.published_on).format('MMMM DD, YYYY HH:mm')}`
+                  : 'Not published'}
```

Comment:
This is not the proper way of doing this. There should be an specific message for each id and not a dynamic one.

``` javascript
{ this.props.selectedContract.published_on &&
  <React.Fragment>
    <FormattedMessage
      id='pipeline.publish-status.yes'
      defaultMessage='Published on'
    /> { moment(this.props.selectedContract.published_on).format('MMMM DD, YYYY HH:mm') }
  </React.Fragment>
}
{ !this.props.selectedContract.published_on &&
  <FormattedMessage
    id='pipeline.publish-status.not'
    defaultMessage='Not published'
  />
}
```

Reply:
Thanks!